caas/kubernetes/provider/k8s.go

```diff
@@ -1634,6 +1656,22 @@ func mergeDeviceConstraints(device devices.KubernetesDeviceParams, resources *co
 	return nil
 }
 
+func mergeConstraint(constraint string, value string, resources *core.ResourceRequirements) error {
+	if resources.Limits == nil {
+		resources.Limits = core.ResourceList{}
+	}
+	resourceName := core.ResourceName(constraint)
+	if v, ok := resources.Limits[resourceName]; ok {
```

Comment:
should we use `Limit` or `Request`?

Reply:
as discussed, we only set max for constraints which is Limit.
Awesome, thanks!

Reply:
It's Limit here because we are setting the max mem/cpu.
We don't yet have a way to specify both using Juju constraint syntax.